tests/pass-dep/concurrency/freebsd-futex.rs

```diff
@@ -0,0 +1,170 @@
+//@only-target: freebsd
+//@compile-flags: -Zmiri-preemption-rate=0
+
+use std::mem::MaybeUninit;
+use std::ptr::{self, addr_of};
+use std::sync::atomic::AtomicU32;
+use std::time::{Duration, Instant};
+use std::{io, thread};
+
+fn wake_nobody() {
+    // TODO: _umtx_op does not return how many threads were woken up
+    // How do i test this?
+}
+fn wake_dangling() {
+    let futex = Box::new(0);
+    let ptr: *const u32 = &*futex;
+    drop(futex);
+
+    // Expect error since this is now "unmapped" memory.
+    unsafe {
+        assert_eq!(
+            libc::_umtx_op(
+                ptr as *const AtomicU32 as *mut _,
+                libc::UMTX_OP_WAKE_PRIVATE,
+                0,
+                ptr::null_mut::<libc::c_void>(),
+                ptr::null_mut::<libc::c_void>(),
+            ),
+            -1
+        );
+        assert_eq!(io::Error::last_os_error().raw_os_error().unwrap(), libc::EFAULT);
+    }
+}
+
+fn wait_wrong_val() {
+    let futex: u32 = 123;
+
+    unsafe {
+        assert_eq!(
+            libc::_umtx_op(
+                ptr::from_ref(&futex).cast_mut().cast(),
+                libc::UMTX_OP_WAIT_UINT_PRIVATE,
+                456,
+                ptr::null_mut::<libc::c_void>(),
+                ptr::null_mut::<libc::c_void>(),
+            ),
+            -1
+        );
+        // man page doesn't document but we set EINVAL for consistency?
+        assert_eq!(io::Error::last_os_error().raw_os_error().unwrap(), libc::EINVAL);
+    }
+}
+
+fn wait_timeout() {
+    let start = Instant::now();
+
+    let futex: u32 = 123;
+
+    // Wait for 200ms, with nobody waking us up early
+    unsafe {
+        assert_eq!(
+            libc::_umtx_op(
+                ptr::from_ref(&futex).cast_mut().cast(),
+                libc::UMTX_OP_WAIT_UINT_PRIVATE,
+                123,
+                &mut libc::timespec { tv_sec: 0, tv_nsec: 200_000_000 } as *mut _ as *mut _,
+                ptr::null_mut::<libc::c_void>(),
+            ),
+            -1
+        );
+        // man page doesn't document but we set EINVAL for consistency
+        assert_eq!(io::Error::last_os_error().raw_os_error().unwrap(), libc::ETIMEDOUT);
+    }
+
+    assert!((200..1000).contains(&start.elapsed().as_millis()));
+}
+fn wait_absolute_timeout() {
```

Comment:
```suggestion

fn wait_absolute_timeout() {
```
In general please take a bit more care of formatting and don't just blindly hope that rustfmt will fix everything. :)